### PR TITLE
{Packaging} Compact aaz folder of `network` module

### PR DESCRIPTION
**Pull request by kairu-ms**
**Related command**

**Description**

This PR implemented an algorithm to compact aaz folder under network modules. It consists of two parts:

**PART1: Merge files**

Merge multiple command files of a command group into one `__cmds.py` file.

Before merge

image

After merge

image

**PART2 Remove duplicated definitions**

There are a lot of duplicated functions defined in aaz. Those functions will be moved in a `_Helper` class and the rest helper class will refer to the definition by using `link_helper`.

image

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Compact network module

**Review by necusjz — APPROVED** [review]

**Review by bebound — APPROVED** [review]

**Comment by bebound** [comment]
Can you add description and result compare about this change?

**Review by calvinhzy — APPROVED** [review]

**Comment by kairu-ms** [comment]
Add PR which contains the compacted aaz https://github.com/Azure/azure-cli/pull/25819

File: scripts/compact_aaz.py [hunk 1]
```diff
@@ -0,0 +1,463 @@
...
+"""
+
+
+class MainModuleCompactor:
+
+    _command_group_pattern = re.compile(r'^class\s+(.*)\(.*AAZCommandGroup.*\)\s*:\s*$')
+    _command_pattern = re.compile(r'^class\s+(.*)\(.*AAZ(Wait)?Command.*\)\s*:\s*$')
+    _command_helper_pattern = re.compile(r'^class\s+(_(.+)Helper)\s*:\s*$')
+    _command_helper_func_pattern = re.compile(r'^\s{4}def (.+)\(cls.*\)\s*:\s*$')
+    _file_end_pattern = re.compile(r'^__all__ = \[.*]\s*$')
```
Comment (jiasli) [inline scripts/compact_aaz.py:129]: We can add some examples here. Without examples, regexes are very hard to read.

File: scripts/compact_aaz.py
Comment (jiasli) [inline scripts/compact_aaz.py:1]: This is a big file. As @bebound stated, I think it is better to have a document explaining how it works overall.

Also, is it possible to integrate the logic into codegen itself?

Comment (kairu-ms) [reply scripts/compact_aaz.py:1]: In order to keep the generated code clean and easy to maintain, the compact should only be done in packaging stage. That's why we don't include it in codegen tool.

**Pull request merged**